pkg/apis/provisioning/v1alpha5/constraints.go

```diff
@@ -47,26 +49,62 @@ type Provider = runtime.RawExtension
 
 // ValidatePod returns an error if the pod's requirements are not met by the constraints
 func (c *Constraints) ValidatePod(pod *v1.Pod) error {
+	p := pod.DeepCopy()
+	// The soft preference may conflict with the requirements.
+	// Remove soft constraints/ preferences
+	if p.Spec.Affinity != nil && p.Spec.Affinity.NodeAffinity != nil {
```

Comment:
We shouldn't need to do this at this level. It should happen at the preferences.go level.

Reply:
Without doing it, pod validation will fail if preferences is conflicting with requirements because at the beginning all preferences are expressed as hard requirements.

Reply:
If they conflict -- it should fail. I don't think we want this to maintain the additional complexity to help dodge user misconfigurations. We intentionally want the logs to complain about the conflict and then relax.

Reply:
Agree. It has been removed.